application_form/tests/test_lottery_haso_services.py

```diff
@@ -296,21 +295,20 @@ def test_winning_does_not_cancel_lower_priority_apartments_if_reserved(
     app_apt2.apartment_reservation.save(update_fields=["state"])
     _distribute_haso_apartments(project_uuid)
     assert list(get_ordered_applications(first_apartment_uuid)) == [app]
-    assert list(get_ordered_applications(second_apartment_uuid)) == [app]
+    assert list(get_ordered_applications(second_apartment_uuid)) == []
     app_apt1.refresh_from_db()
     app_apt2.refresh_from_db()
     assert app_apt1.apartment_reservation.state == ApartmentReservationState.RESERVED
-    assert app_apt2.apartment_reservation.state == ApartmentReservationState.RESERVED
+    assert app_apt2.apartment_reservation.state == ApartmentReservationState.CANCELED
 
```

Comment:
IMO we should also test at least one state that should not be cancelled. Or maybe parametrize the test and test all of them.

Reply:
I picked only one state other than SUBMITTED and RESERVED to add here. Adding more states does not mean much in this test case and it might increase the test time, which is already slow atm :(